src/include/daos/common.h

```diff
@@ -799,6 +799,36 @@ daos_unparse_ctype(daos_cont_layout_t ctype, char *string)
 	}
 }
 
+static inline void
+daos_anchor_set_flags(daos_anchor_t *anchor, uint32_t flags)
+{
+	anchor->da_flags = flags;
+}
+
+static inline uint32_t
+daos_anchor_get_flags(daos_anchor_t *anchor)
+{
+	return anchor->da_flags;
+}
```

Comment:
I think we may need to cleanup daos_anchor_t and anchor APIs a bit, current daos_anchor_t exposes too much server internal stuff.

I think there are two ways to cleanup the anchor:
1. Make daos_anchor_t an opaque data structure, user always needs to allocate/free anchor by daos_anchor_init/fini(), or;
2. Get rid of "da_shard", "da_flags" & "da_sub_anchors" from daos_anchor_t, make them as part of "da_buf" or another opaque field.

Any thoughts?

Reply:
dynamically allocation anchor is a significant API change, it may break a lot of things.
I think the direction we should choose is like daos_event_t, just add bytes for space and convert them to data structure internally. We can do this in follow-on patch.

Reply:
Sounds good.